php/docker/templates/nginx-conf.d/90-metrics.conf

```
@@ -0,0 +1,87 @@
+{{- $metricsEnabled := default "true" .Env.STACK_METRICS_ENABLED }}
+
+{{- if isTrue $metricsEnabled }}
+
+lua_shared_dict prometheus_metrics 10M;
+
+{{- $siteName := default "" .Env.STACK_SITE_NAME }}
+
+init_worker_by_lua_block {
+    prometheus = require("prometheus").init("prometheus_metrics")
+    metric_requests = prometheus:counter(
+        "nginx_http_requests_total",
+        "Number of HTTP requests",
+        {"host", "site_name", "status", "cache_status"}
+    )
+    metric_request_time = prometheus:histogram(
+        "nginx_http_request_duration_seconds",
+        "HTTP request request duration",
+        {"host", "site_name"}
+    )
+    metric_connections = prometheus:gauge(
+        "nginx_http_connections",
+        "Number of HTTP connections",
+        {"host", "site_name", "state"}
+    )
+}
+
+log_by_lua_block {
+    metric_requests:inc(
+        1, {
+            ngx.var.server_name,
+            {{ $siteName }}
+            ngx.var.status,
+            string.lower(ngx.resp.get_headers()["x-cache-fetch"] or "disabled")
+        }
+    )
+
+    metric_request_time:observe(
+        tonumber(ngx.var.request_time), {
+            ngx.var.server_name,
+            {{ $siteName }}
+        }
+    )
+}
+
+server {
+    listen {{ default "9145" .Env.STACK_METRICS_PORT | atoi }};
+
+    allow 127.0.0.1;
+    {{- if (default "10.0.0.0/8" .Env.STACK_METRICS_WHITELIST_CIDR) }}
+    allow {{ default "10.0.0.0/8" .Env.STACK_METRICS_WHITELIST_CIDR }};
+    {{- end }}
+    deny  all;
+
+    location /metrics/wordpress {
```

What do you think about prefixing with `/-/` ?
```suggestion
    location /-/metrics/wordpress {
```

---

Done. Should we also define them from env so that you can reuse those vars when defining the prometheus targets?